rdf_differ/diff_adapter.py

```diff
@@ -0,0 +1,344 @@
+        """
+            Get number of deleted triples in the given dataset.
+        :param dataset_name: The dataset identifier. This should be short alphanumeric string uniquely
+        identifying the dataset
+        :return: deleted triples count
+        :rtype: int, int
+        """
+        query_result, status = self.execute_query(dataset_name=dataset_name,
+                                                  sparql_query=SKOS_HISTORY_PREFIXES + QUERY_DELETIONS_COUNT)
+        return self._extract_deletion_count(query_result), status
+
+    def diff_description(self, dataset_name: str) -> tuple:
+        """
+            Provide a generic description of the dataset.
+        :param dataset_name: The dataset identifier. This should be short alphanumeric string uniquely
+        identifying the dataset
+        :return:
+            diff description:
+            * datasetURI = dataset/scheme URI,
+            * version history graph = the graph that stores the structure of the calculated diffs
+            * current version graph = the graph that corresponds to the latest version of the dataset,
+            * datasetVersions = list of loaded dataset versions as declared by the datasets themselves,
+            * versionIds = list of versionIds as provided in the configurations file
+            * versionNamedGraphs = named graphs where the versions of datasets are loaded
+        :rtype: dict, int
+        """
+        query_result, status = self.execute_query(dataset_name=dataset_name,
+                                                  sparql_query=SKOS_HISTORY_PREFIXES + QUERY_DATASET_DESCRIPTION)
+
+        return self._extract_dataset_description(response=query_result, dataset_id=dataset_name,
+                                                 query_url=self.make_sparql_endpoint(dataset_name)), status
+
+    def delete_dataset(self, dataset_name: str) -> tuple:
+        """
+            Delete the dataset from the Fuseki store
+        :param dataset_name: The dataset identifier. This should be short alphanumeric string uniquely
+        identifying the dataset
+        :return: text and status of the deleted dataset
+        :rtype: text, int
+        """
+        response = requests.delete(urljoin(self.triplestore_service_url, f"/$/datasets/{dataset_name}"),
+                                   auth=HTTPBasicAuth('admin', 'admin'))
+
+        return response.text, response.status_code
+
+    def list_datasets(self) -> tuple:
+        """
+            Get the list of the dataset names from the Fuseki store.
+        :return: the list of the dataset names
+        :rtype: list, int
+        """
+        response = requests.get(urljoin(self.triplestore_service_url, "/$/datasets"),
+                                auth=HTTPBasicAuth('admin', 'admin'))
+
+        if response.status_code != 200:
+            return {
+                       'error': f"Fuseki server request ({response.url}) got response {response.status_code}"
```

Comment:
can writing the `response.url` in the error could expose to security vulnerabilities?

Reply:
we take no security considerations at this point as it is supposed to be an internal/closed system